cmd/xl-storage.go

```diff
@@ -890,6 +890,13 @@ func (s *xlStorage) DeleteVersion(ctx context.Context, volume, path string, fi F
 		return err
 	}
 
+	var transitionStatus string
+	xfi, err := xlMeta.ToFileInfo(volume, path, fi.VersionID)
```

Comment:
@kp, this is the cause of the `mc rm` failure - fi.VersionID would come in with a random uuid to create a "Deleted" version entry if there is a rm operation. This check has to be thought out better